sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java

```diff
@@ -1412,6 +1405,84 @@ public List<PersonIndexDto> getIndexList(PersonCriteria criteria, Integer first,
 		return persons;
 	}
 
+	private List<Long> getIndexListIds(PersonCriteria criteria, Integer first, Integer max, List<SortProperty> sortProperties) {
+
+		long startTime = DateHelper.startTime();
+		final CriteriaBuilder cb = em.getCriteriaBuilder();
+		final CriteriaQuery<Tuple> cq = cb.createTupleQuery();
+		final Root<Person> person = cq.from(Person.class);
+
+		final PersonQueryContext personQueryContext = new PersonQueryContext(cb, cq, person);
+		final PersonJoins personJoins = personQueryContext.getJoins();
+		personJoins.configure(criteria);
+
+		List<Selection<?>> selects = new ArrayList<>();
+		selects.add(person.get(Person.ID));
+
+		if (sortProperties != null && sortProperties.size() > 0) {
```

Comment:
This `sortproperties` block is completely duplicated and should by extracted into a separate method.

Reply:
Done.